### PR TITLE
Jenkins plugin ImageStream SCM extended tests

### PR DESCRIPTION
**Pull request by csrwng**
Adds extended tests to validate the functionality of the Jenkins plugin imagestream SCM and corresponding DSL step.

### CODE REVIEW

**Comment by csrwng** [comment]
Depends on https://github.com/openshift/origin/pull/11252

**Comment by csrwng** [comment]
[testextended][extended:core(imagestream SCM)]

**Comment by csrwng** [comment]
I've now rebased on the latest master
@gabemontero @jupierce ptal

File: test/extended/image_ecosystem/plugin.go [hunk 1]
```diff
@@ -770,6 +771,65 @@ var _ = g.Describe("[image_ecosystem][jenkins][Slow] openshift pipeline plugin",
...
+
+			// Now create a new imagestream tag and expect a job to be kicked off
+			// that will create a new tag in the current namespace
+			g.By("creating an imagestream tag in the current project")
+			oc.Run("tag").Args("openshift/jenkins:latest", fmt.Sprintf("%s/testimage:v1", oc.Namespace())).Execute()
+
+			// Now wait after the image has been tagged for the Jenkins job to run
+			// and create the new imagestream/tag
+			g.By("verifying that the job ran by looking for the resulting imagestream tag")
+			err := wait.Poll(5*time.Second, 10*time.Minute, func() (bool, error) {
```
Comment (jupierce) [inline test/extended/image_ecosystem/plugin.go]: You might consider using framework.go:WaitForAnImageStreamTag to keep things DRY. Until #11570 merges, it only waits 1 minute, but you can bump it to 10 and I will rebase #11570 if you merge first.

File: test/extended/image_ecosystem/plugin.go [hunk 2]
```diff
@@ -770,6 +771,65 @@ var _ = g.Describe("[image_ecosystem][jenkins][Slow] openshift pipeline plugin",
...
 
+		g.It("jenkins-plugin test imagestream SCM", func() {
+			jobName := "test-imagestream-scm"
+			g.By("creating a jenkins job with an imagestream SCM")
+			data := j.readJenkinsJob("imagestream-scm-job.xml", oc.Namespace())
+			j.createItem(jobName, data)
+
+			// Because polling is enabled, a job should start automatically and fail
+			// Wait long enough for nothing to be running
+			time.Sleep(3 * time.Minute)
```
Comment (jupierce) [inline test/extended/image_ecosystem/plugin.go]: Perhaps a wait.Poll{ getResource(...) } on the current job colors could eliminate the static wait?
http://stackoverflow.com/questions/14843874/from-jenkins-how-do-i-get-a-list-of-the-currently-running-jobs-in-json

Comment (csrwng) [inline test/extended/image_ecosystem/plugin.go]: sounds better, will do

**Comment by gabemontero** [comment]
Noting to add to the discussion that has already taken place.  IGTM.

**Comment by csrwng** [comment]
- Addressed comments
- refactored tests to a single function since they're essentially the same except for the job being used. 
- Fixed an issue with the TEST_ONLY=1 case, where the artifact directory was not getting set
- Added a function to specify the timeout when waiting on an ImageStreamTag (@jupierce I thought this may be easier to do if for the most part we want a 1 minute timeout), plus less code to change.

**Comment by csrwng** [comment]
For tests to pass, this will require this fix from the jenkins-plugin: https://github.com/openshift/jenkins-plugin/pull/100

**Comment by csrwng** [comment]
@gabemontero @jupierce the extended tests now pass, is this ready for merge?
@bparees fyi

**Comment by gabemontero** [comment]
interesting ... we've seen some permission issues with projects and the service accounts start occurring over the weekend, but they did not occur with your run .... I see there is no merge conflicts, but did you rebase today?  If not perhaps that is why you did not see these new issues with the extended tests.

That curiosity is not a blocker for me though .... IGTM

**Comment by csrwng** [comment]
Yes, I did rebase today. I'm only focusing on the new tests introduced here, so maybe that's why those issues didn't show up?

**Comment by gabemontero** [comment]
Yeah perhaps it does not show up with just your test.

**Comment by bparees** [comment]
@gabemontero @csrwng 
rebases don't matter, the bot always tests by merging your changes into master before running the test.

(rebases *do* matter for our image repos, because we use the PR tester for those repos and it's less sophisticated.  but for stuff that's managed by the bot, you should never need to rebase to get the bot to test w/ the latest master+your changes)

**Comment by bparees** [comment]
[merge]

**Pull request merged**